src/num.rs

```diff
@@ -706,13 +762,13 @@ macro_rules! float_bin_search {
                     };
                     let new_mid = self.lo + interval/2.0;
 
-                    let new_mid = if new_mid == self.curr || 0.0 == interval {
+                    let new_mid = if nearly_eq(new_mid, self.curr) || 0.0 == interval {
```

Comment:
I'm fairly confident the strict equality is correct. Was this just changed because of the clippy lint?

Reply:
Yes; the tests do pass however :)

So... could we document why the strict equality is correct here?